crates/cxx-qt-gen/src/parser/cxxqtdata.rs

```diff
@@ -71,8 +74,62 @@ impl ParsedCxxQtData {
                 if let Some(index) =
                     attribute_find_path(&qobject_struct.attrs, &["cxx_qt", "qobject"])
                 {
+                    let attrs_map = attribute_tokens_to_map::<Ident, LitStr>(
+                        &qobject_struct.attrs[index],
+                        AttributeDefault::Some(|span| LitStr::new("", span)),
+                    )?;
+
+                    // Parse QML metadata
+                    let mut qml_metadata = None;
+                    let qml_uri = attrs_map.get(&quote::format_ident!("qml_uri"));
+                    let qml_version = attrs_map.get(&quote::format_ident!("qml_version"));
+                    let qml_name = attrs_map.get(&quote::format_ident!("qml_name"));
+                    let qml_uncreatable = attrs_map.get(&quote::format_ident!("qml_uncreatable"));
+                    let qml_singleton = attrs_map.get(&quote::format_ident!("qml_singleton"));
+                    match (qml_uri, qml_version) {
+                        (Some(qml_uri), Some(qml_version)) => {
+                            let qml_version = qml_version.value();
+                            let version_parts: Vec<_> = qml_version.split('.').collect();
+                            let version_major = version_parts[0]
+                                .parse()
+                                .expect("Could not parse major version from qml_version");
+                            let version_minor =
+                                version_parts.get(1).unwrap_or(&"0").parse().unwrap_or(0);
+
+                            let name = match qml_name {
+                                Some(qml_name) => qml_name.value(),
+                                None => qobject_struct.ident.to_string(),
+                            };
+
+                            qml_metadata = Some(QmlElementMetadata {
+                                uri: qml_uri.value(),
+                                name,
+                                version_major,
+                                version_minor,
+                                uncreatable: qml_uncreatable.is_some(),
+                                singleton: qml_singleton.is_some(),
+                            });
+                        }
+                        (Some(_), None) => panic!("qml_uri specified but no qml_version specified"),
```

Comment:
Instead of panicing here, return a syn::Result instead with a syn::Error:
ex.:
```
use syn::Error;
Err(Error::new(
    /*qml_uri attribute span*/,
    "qml_uri specified but no qml_version specified",
));
```
Same with the other cases.